lib/service/contract_base.rb

```diff
@@ -8,8 +8,29 @@ class Service::ContractBase
 
   delegate :slice, :merge, to: :to_hash
 
+  class << self
+    def attribute(name, cast_type = nil, **options, &block)
+      return super(name, cast_type, **options) unless block_given?
+
+      nested_contract_class =
+        Class.new(Service::ContractBase) do
+          define_singleton_method(:name) { "#{name}_contract".classify }
```

Comment:
`ActiveModel` needs a name when generating errors. This is usually based on the object’s class, but here we’re just generating anonymous classes, so we’re giving them a name.